src/app/components/main-page/main-page.component.ts

```diff
@@ -15,30 +18,61 @@ export class MainPageComponent implements OnInit {
   faSearch = faSearch;
   faUser = faUser;
   faHome = faHome;
+
   isSearchCollapsed = true;
   isAtHomePage = false;
   isAtProfilePage = false;
+  user: User;
 
   constructor(
     private firebaseAuth: AngularFireAuth,
-    private router: Router) {
+    private router: Router,
+    private eventsService: EventsService,
+    private toastr: ToastrService,
+    private firebaseStorage: AngularFireStorage) {
 
-      router.events.subscribe((event: RouterEvent) => {
-        if (event instanceof NavigationEnd) {
-          this.isAtHomePage = router.url === '/home';
-          this.isAtProfilePage = router.url === '/';
-          // TODO possibly also set website tile here
-        }
-      });
+    router.events.subscribe((event: RouterEvent) => {
+      if (event instanceof NavigationEnd) {
+        this.isAtHomePage = router.url === '/home';
+        this.isAtProfilePage = router.url === '/';
+        // TODO possibly also set website tile here
+      }
+    });
 
-    }
+  }
 
   ngOnInit() {
     this.firebaseAuth.auth.onAuthStateChanged(user => {
       if (!user) {
         this.router.navigateByUrl('/login');
       }
     });
+
+    this.firebaseAuth.user.subscribe(
+      user => this.user = user
+    );
+
+    this.eventsService.currentPhotosToDelete.subscribe(x => {
+      this.deletePhotos(x);
+    });
```

Comment:
I think we might need to create subscription variables here and assign each subscription happening here to a subscription variable and then unsubscribe from it in ngDestroy.

Reply:
done